test/env/data.go

```diff
@@ -288,3 +290,32 @@ func EventuallyBundleHasSyncedAllNamespacesStartsWith(ctx context.Context, cl cl
 		ctx, cl, bundleName, startingData,
 	).Should(BeNil(), fmt.Sprintf("checking bundle %s has synced to all namespaces with correct starting data", bundleName))
 }
+
+// CheckJKSFileSynced ensures that the given JKS data
+func CheckJKSFileSynced(jksData []byte, expectedPassword string, expectedCertPEMData string) error {
+	reader := bytes.NewReader(jksData)
+
+	ks := jks.New()
+
+	err := ks.Load(reader, []byte(expectedPassword))
+	if err != nil {
+		return err
+	}
+
+	expectedCertList, err := util.ValidateAndSplitPEMBundle([]byte(expectedCertPEMData))
+	if err != nil {
+		return fmt.Errorf("invalid PEM data passed to CheckJKSFileSynced: %s", err)
+	}
+
+	// TODO: check that the cert content matches expectedCertPEMData exactly, not just
+	// that the count is the same
```

Comment:
note: implementing this would take a bit more work. It's doable but I didn't want this PR to balloon so I skipped it for this fix.